File: Node/encrypted-pdf-endpoint.js
/**
The /encrypted-pdf endpoint can take a single PDF file or id as input.

This sample demonstrates setting the open password to 'password'.

Import fetch
 */
import fetch, { FormData, fileFromSync } from "node-fetch";

// Append formdata here
let formdata = new FormData();
formdata.append("file", fileFromSync("../Sample_Input/toEncrypt.pdf"));
formdata.append("new_open_password", "password");
formdata.append("output", "example_encryptedPdf_out");

let requestOptions = {
  method: "POST",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Place your api key here
  },
  body: formdata,
  redirect: "follow",
};

// Define URL and submit request
fetch("https://api.pdfrest.com/encrypted-pdf", requestOptions)
  .then((response) => response.text())
  .then((result) => console.log(result))
  .catch((error) => console.log("error", error));

// If you would like to download the file instead of getting the JSON response, please see the 'get-resource-id-endpoint.js' sample.

File: Python/encrypted-pdf-endpoint.py
import requests

# Set the API endpoint URL
url = "https://api.pdfrest.com/encrypted-pdf"

# Define the payload data for the request
payload = {
    'new_open_password': 'new_example_pw',
    'output': 'pdfrest_encrypted_pdf'
}

# Specify the file to be encrypted
files = [
    ('file', ('file', open('/path/to/file', 'rb'), 'application/octet-stream'))
]

# Set the headers, including the API key
headers = {
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'  # Replace with your API key
}

# Make a POST request to the API endpoint
response = requests.request(
    "POST", url, headers=headers, data=payload, files=files)

# Print the response content
print(response.text)
